### PR TITLE
Profile Screen

### PR DESCRIPTION
**Pull request by MadaMada08**
## Summary

This PR introduces the **ProfileScreen** to our app and integrates it into the navigation flow.  
Users can now view their profile details (name, email, address, and default vet for farmers) and log out or navigate back to the Overview screen from the Profile screen.

---

### ProfileScreen
- I implemented the `ProfileScreen` composable showing:
  - User’s **name**, **email**, **address**, and **password (masked)**.
  - **Default vet field** shown only for farmers.
  - A **round profile image placeholder icon** centered above the name. *
  - An **edit profile** icon button (currently inactive).
  - A **logout** button in the top bar.
  - A **back arrow** to return to the Overview screen.

(I realized we might want the profile image variable to the User class in the near future.)*

### Navigation
- I added a new **Profile route** (`Screen.Profile`) to the app’s navigation.
- Integrated the ProfileScreen into the **Overview → Profile → Overview** navigation flow.
  - Clicking the **profile icon** on the top-left of the Overview screen now navigates to the Profile screen.
  - Clicking the **back arrow** on the Profile screen returns to the Overview screen.
  - Clicking **logout** navigates to the Auth screen.

### OverviewScreen Update
- I added a new **profile icon button** to the **TopAppBar** on the left side.
- Connected it to the navigation graph via `navigationActions.navigateTo(Screen.Profile)`.

---

## Testing

- I added a comprehensive **Compose UI test suite** for `ProfileScreen`, using **test tags** for stable selectors.
- Achieved **~84% line coverage** on new code.
- Tests include verification of:
  - All key UI components (top bar, buttons, text fields, etc.)
  - Conditional visibility (default vet only for farmers)
  - Safe fallbacks for null user
  - Functional callbacks for edit, back, and logout

---

## Refactor / Code Improvements

- I moved the `address` field from `Farmer` and `Vet` to the **base `User` class** for better consistency and shared structure across user roles.

---

## Next Steps/ Note

- I will start (and finish hopefully) implementing **EditProfileScreen** tomorrow.
- Once the code generation logic is added to the project, I’ll wire it up behind the **Generate Code** button on `ProfileScreen`.
- For once, this PR is openned with a finished version of the code, so if no errors are found it should be possible to merge it tonight.

### CODE REVIEW

File: AgriHealth-Alert-main/app/src/androidTest/java/com/android/agrihealth/ui/profile/ProfileScreenTest.kt
Comment (FrancoAdam) [inline AgriHealth-Alert-main/app/src/androidTest/java/com/android/agrihealth/ui/profile/ProfileScreenTest.kt]: I think this test suite is great with many edge cases covered, I would advice you to change the constant strings you use for test tags by a companion object that stores these constant strings so it's easier for others to test your components and makes the tests less error prone during the writing phase.

File: AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/ui/profile/ProfileScreen.kt [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                  when (user) {
+                    is Farmer -> user.address
+                    is Vet -> user.address
+                    else -> null
+                  }
+
+              OutlinedTextField(
+                  value = address?.toString() ?: "",
+                  onValueChange = {},
+                  label = { Text("Address") },
```
Comment (FrancoAdam) [inline AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/ui/profile/ProfileScreen.kt]: Naming this field "Address" could be a bit confusing for users, I would suggest naming this field something that makes this more clearly work related.

File: AgriHealth-Alert-main/app/src/androidTest/java/com/android/agrihealth/ui/profile/ProfileScreenTest.kt [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+  private fun setScreen(vm: UserViewModel) {
+    composeTestRule.setContent {
+      MaterialTheme {
+        ProfileScreen(userViewModel = vm, onGoBack = {}, onLogout = {}, onEditProfile = {})
+      }
+    }
+  }
+
+  // Tests suite: 84% coverage
```
Comment (FrancoAdam) [inline AgriHealth-Alert-main/app/src/androidTest/java/com/android/agrihealth/ui/profile/ProfileScreenTest.kt]: This comment might be accurate now but won't be as the app develops, I suggest removing it.

File: AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/MainActivity.kt [hunk 3]
```diff
@@ -147,6 +148,18 @@ fun AgriHealthApp() {
                 viewModel = viewModel,
                 reportId = reportId)
           }
+      composable(Screen.Profile.route) {
+        ProfileScreen(
+            userViewModel = userViewModel,
+            onGoBack = { navigationActions.goBack() },
+            onLogout = {
+              Firebase.auth.signOut()
```
Comment (FrancoAdam) [inline AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/MainActivity.kt]: Firebase.auth.signout() isn't enough to cleanly log out the user now since we work with a credential manager for google accounts on top of our older accounts, check how the overview view model handles this for reference.

File: AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/ui/profile/ProfileScreen.kt [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+                  enabled = false,
+                  modifier = Modifier.fillMaxWidth().testTag("EmailField"))
+
+              Spacer(modifier = Modifier.height(12.dp))
+
+              // Password
+              OutlinedTextField(
+                  value =
+                      "********", // I think we are missing the passwords in the User model, should
+                  // we modify this ?
```
[the remark this answers is not in the document]
Comment (MadaMada08) [reply AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/ui/profile/ProfileScreen.kt]: Agreed I will change the comment

File: AgriHealth-Alert-main/app/src/androidTest/java/com/android/agrihealth/ui/profile/ProfileScreenTest.kt (hunk 2 above)
Comment (MadaMada08) [reply AgriHealth-Alert-main/app/src/androidTest/java/com/android/agrihealth/ui/profile/ProfileScreenTest.kt]: Done!

File: AgriHealth-Alert-main/app/build.gradle.kts [hunk 5]
```diff
@@ -151,6 +151,7 @@ dependencies {
     implementation(libs.compose.ui.graphics)
     // Material Design 3
     implementation(libs.compose.material3)
+    implementation("androidx.compose.material:material-icons-extended:<version>")
```
Comment (nils-caspar) [inline AgriHealth-Alert-main/app/build.gradle.kts]: The intended and clean way to add this kind of line is to use `gradle/libs.versions.toml`. Android studio should highlight this line and automatically fix this for you, then you have to put the right version number in the variable `materialIconsExtended` (current latest version is 1.7.8)
But anyway it looks like this isn't used anywhere in the code?

File: AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/MainActivity.kt [hunk 6]
```diff
@@ -156,6 +157,21 @@ fun AgriHealthApp(
...
+
+        ProfileScreen(
+            userViewModel = userViewModel,
+            onGoBack = { navigationActions.goBack() },
+            onLogout = {
+              overviewViewModel.signOut(credentialManager)
+              navigationActions.navigateToAuthAndClear()
+            },
+            onEditProfile = {
+              // Later we will add edit profile functionality
```
Comment (nils-caspar) [inline AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/MainActivity.kt]: you can add `TODO:` to be able to find this line more easily, that's what I do anyway since Android studio highlights those

File: AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/ui/profile/ProfileScreen.kt [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = {
+              Text(
+                  text = "Profile - ${userRole.displayString()}",
+                  style = MaterialTheme.typography.titleMedium)
+            },
+            navigationIcon = {
+              IconButton(onClick = onGoBack, modifier = Modifier.testTag("GoBackButton")) {
```
Comment (nils-caspar) [inline AgriHealth-Alert-main/app/src/main/java/com/android/agrihealth/ui/profile/ProfileScreen.kt]: There's a test tag for that in `NavigationTestTags`
